Arrowgene.Ddon.GameServer/Characters/AreaRankManager.cs

```diff
@@ -0,0 +1,159 @@
+using Arrowgene.Ddon.Server;
+using Arrowgene.Ddon.Server.Network;
+using Arrowgene.Ddon.Shared.Entity.PacketStructure;
+using Arrowgene.Ddon.Shared.Entity.Structure;
+using Arrowgene.Ddon.Shared.Model;
+using Arrowgene.Ddon.Shared.Model.Quest;
+using Arrowgene.Logging;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Linq;
+
+namespace Arrowgene.Ddon.GameServer.Characters
+{
+    public class AreaRankManager
+    {
+        private static readonly ServerLogger Logger = LogProvider.Logger<ServerLogger>(typeof(ClanManager));
+
+        private readonly DdonGameServer Server;
+
+        public AreaRankManager(DdonGameServer server)
+        {
+            Server = server;
+        }
+
+        public List<CDataRewardItemInfo> GetSupplyRewardList(QuestAreaId areaId, uint rank, uint points)
+        {
+            List<CDataRewardItemInfo> list = new();
+
+            var areaSupplies = Server.AssetRepository.AreaRankSupplyAsset.Where(x => x.AreaId == areaId);
+            if (!areaSupplies.Any())
+            {
+                Logger.Error($"No valid asset for {areaId} found in AreaRankSupply asset.");
+                return new();
```

Comment:
Would we want to silently fail to the user or would we want to throw an error instead